ext/vulnsrc/redhat/redhat.go

```diff
@@ -189,8 +214,15 @@ func CollectVulnerabilities(advisoryDefinitions []ParsedAdvisory, ovalDoc OvalV2
 // CollectVulnsForAdvisory - get the set of vulns for the given advisory (full doc must also be passed, for the states/tests/objects references)
 func CollectVulnsForAdvisory(advisoryDefinition ParsedAdvisory, ovalDoc OvalV2Document) (vulnerabilities []database.VulnerabilityWithAffected) {
 	// first, check the advisory severity
-	if (IsSignificantSeverity(advisoryDefinition.Metadata.Advisory.Severity)) {
+	if IsSignificantSeverity(advisoryDefinition.Metadata.Advisory.Severity) && IsSupportedDefinitionType(advisoryDefinition.Class) {
 		for _, cve := range advisoryDefinition.Metadata.Advisory.CveList {
+			// first, make sure the pending vulnerabilities set doesn't already have this vulnerability def (since they can appear in multiple manifest entries)
+			if pendingVulnNames[cve.Value+" - "+ParseRhsaName(advisoryDefinition)] {
```

Comment:
I don't think this is a reliable solution for how to remove duplicated items... If there is an advisory that ships 2 packages -first for RHEL 6 and other for RHEL 7 then you end up having the same CVE + RHSA in 2 different oval files, but with different rpms and CPEs and your code would process a first one, but ignore the other one.

Reply:
Here is the example of such an advisory: https://access.redhat.com/errata/RHSA-2019:1619

Reply:
Please create a new `deduplicate` function that gets all vulnerabilities from your oval parser then it finds duplicated items based on vulnerability name and merges its affected packages and keeps only a unique one.

With this algorithm, you don't skip any advisory and at the same time filter out duplicated affected packages.

Reply:
@Allda fyi I've updated with a dedup algorithm based on your suggestion, currently testing, will push once it's validated

Reply:
tested and pushed with updated dedup strategy